src/python_testing/mdns_discovery/mdns_discovery.py

```diff
@@ -0,0 +1,274 @@
+            weight=service_info.weight,
+            host_ttl=service_info.host_ttl,
+            other_ttl=service_info.other_ttl
+        )
+
+        return mdns_service_info
+
+    def _get_dut_operational_service_name(self) -> str:
+        """
+        Constructs the operational service name for the DUT.
+
+        This method generates a service name string based on the DUT's node ID and the compressed
+        fabric ID, following the operational service naming convention.
+
+        Returns:
+            str: A string representing the operational service name.
+        """
+        node_id = self._tc.dut_node_id
+        compressed_fabric_id = self._tc.default_controller.GetCompressedFabricId()
+        service_name = f'{compressed_fabric_id:016X}-{node_id:016X}.{MdnsType.OPERATIONAL.value}'
+        return service_name
+
+    def _get_service_info(self, mdns_type: MdnsType) -> List[Dict[str, any]]:
+        """
+        This method returns the discovered service information for a specified mDNS type.
+        It raises an exception if the discovery process has not been performed before calling this method.
+
+        Args:
+            mdns_type (MdnsType): The mDNS type for which to retrieve the service info.
+
+        Returns:
+            list: A list of discovered services of the specified mDNS type.
+
+        Raises:
+            DiscoveryNotPerformedError: If discovery has not been performed before calling this method.
+        """
+        if not self._discovery_performed:
+            raise DiscoveryNotPerformedError()
+
+        return self._discovered_services[mdns_type.name]
+
+    def get_commissioner_service_info(self) -> List[Dict[str, any]]:
+        """
+        Retrieves the service information for Commissioner services.
+
+        Returns:
+            list: A list of discovered Commissioner services.
+        """
+        return self._get_service_info(MdnsType.COMMISSIONER)
+
+    def get_commissionable_service_info(self) -> List[Dict[str, any]]:
+        """
+        Retrieves the service information for Commissionable services.
+
+        Returns:
+            list: A list of discovered Commissionable services.
+        """
+        return self._get_service_info(MdnsType.COMMISSIONABLE)
+
+    def get_operational_service_info(self) -> List[Dict[str, any]]:
+        """
+        Retrieves the service information for Operational services.
+
+        Returns:
+            list: A list of discovered Operational services.
+        """
+        return self._get_service_info(MdnsType.OPERATIONAL)
+
+    def get_border_router_service_info(self) -> List[Dict[str, any]]:
+        """
+        Retrieves the service information for Border Router services.
+
+        Returns:
+            list: A list of discovered Border Router services.
+        """
+        return self._get_service_info(MdnsType.BORDER_ROUTER)
+
+    async def _async_close(self) -> None:
+        assert self._zc is not None
```

Comment:
rather than asserts, can you just check if this is none? Then this doesn't become dangerous to call at the wrong time.